File: codewars/7kyu/disarium-number-(special-numbers-series-#3)/dduck-disarium-number-(special-numbers-series-#3).js
// https://www.codewars.com/kata/5a53a17bfd56cb9c14000003/

const disariumNumber = (n) => {
  const result = String(n)
    .split('')
    .map((v, i) => Number(v) ** (i + 1))
    .reduce((a, c) => (a + c));
  
  return result === n ? 'Disarium !!' : 'Not !!';
};

console.log(disariumNumber(89), 'Disarium !!');
console.log(disariumNumber(564), 'Not !!');
console.log(disariumNumber(1024), 'Not !!');
console.log(disariumNumber(135), 'Disarium !!');
console.log(disariumNumber(136586), 'Not !!');

File: codewars/7kyu/disarium-number-(special-numbers-series-#3)/dduck-disarium-number-(special-numbers-series-#3).py
# https://www.codewars.com/kata/5a53a17bfd56cb9c14000003/

from unittest import TestCase


def disarium_number(number):
    if number == sum([int(v) ** (i + 1) for i, v in enumerate(str(number))]):
        return "Disarium !!"
    return "Not !!"


TestCase().assertEqual(disarium_number(89), "Disarium !!")
TestCase().assertEqual(disarium_number(518), "Disarium !!")
TestCase().assertEqual(disarium_number(1024), "Not !!")
